chore(ui): update onboarding copy

abbey-titcomb opened the pull request:
- changed onboarding copy
- edited manage remotes search bar + button text
- edited copy in search bar
- edited copy + changed order in Settings

TODO:

- [x] Replace user URN with peerID in Success screen and User Profile page
- [x] ~~Decide on `peer address` vs `peerID`~~ Replace PeerID w/ Device ID & projectID w/ Radicle ID
- [x] ~~Add a "helper hinter" to Manage Remotes modal (similar to the create project flow)~~ Creating a separate issue

rudolfs commented on ui/Modal/ManagePeers.svelte:
```diff
@@ -92,7 +92,7 @@
           hint="v"
           bind:value={newPeer}
-          placeholder="Paste a remote address here"
+          placeholder="Paste a peer address here..."
```
```suggestion
          placeholder="Paste a peer address here…"
```

We use ellipses everywhere else.

rudolfs commented on ui/Screen/Onboarding/Success.svelte:
```diff
@@ -45,8 +45,9 @@
+      This is a <span class="typo-text-bold">peer address</span>! It's unique to
+      you and this computer. You can find it at any time in your Settings page.
+      You'll need to share it with others to collaborate.
     </p>
```
We should also change the code above to actually show a `PeerId` instead of our own `URN`.

rudolfs reviewed (commented):
Had a preliminary peek. 👀 

Would be great if you would set up commit signing: https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/signing-commits.

Lmk if you need help with that!

rudolfs commented:
> Add tooltip to Manage Remotes modal?

Is it these tooltips?
<img width="50%" alt="Screenshot 2020-11-09 at 09 37 47" src="https://user-images.githubusercontent.com/158411/98518136-45a3a980-226f-11eb-88e0-2700f9742a8d.png">

If so, I had them there initially, but had to remove them after a discussion with @radicle-dev/product.

FintanH commented on ui/Modal/ManagePeers.svelte:
```diff
@@ -92,7 +92,7 @@
           hint="v"
           bind:value={newPeer}
-          placeholder="Paste a remote address here"
+          placeholder="Paste a peer address here..."
```
Are we using the word "address"? Or should it be `peerID` -- or whatever casing we're going for?

abbey-titcomb replied:
> > Add tooltip to Manage Remotes modal?
> 
> Is it these tooltips?
> <img alt="Screenshot 2020-11-09 at 09 37 47" width="50%" src="https://user-images.githubusercontent.com/158411/98518136-45a3a980-226f-11eb-88e0-2700f9742a8d.png">
> 
> If so, I had them there initially, but [had to remove them](https://github.com/radicle-dev/radicle-upstream/pull/1155/files#diff-52c26d24276defc75692d1883dd7f53ffdde278c62b6c6a5ce0fd15c4e904e75L57) after a discussion with @radicle-dev/product.

Ok, good to know. This was something I added so it could be discussed. Seems like we can make the Manage Remotes flow easier to understand with a tooltip, but I'll check out the previous conversation on it. thnx

rudolfs replied:
It now [says](https://github.com/radicle-dev/radicle-upstream/pull/1206/files#diff-cda325a03becd89efda336f88f202071175a39195079244f352ddb39bf04e535R95): `Paste a Device ID here…`.

FintanH commented on ui/Screen/Onboarding/Success.svelte:
```diff
@@ -45,8 +45,9 @@
       color: var(--color-foreground-level-6);">
-      This is your handle — it's unique to you!<br /> Click to copy it and share
-      with others so that they can find you.
+      This is a <span class="typo-text-bold">peer address</span>! It's unique to
```
Ah ok, seems we're going with "address" cool cool cool

abbey-titcomb replied:
It's either going to be peerID or peer address pending my convo with @cloudhead today. Just was trying out peer address in the copy to see how it feels

FintanH replied:
I'd personally advocate for "PeerId" since that's how it will show up in radicle-link documentation. And the fact that an address in the network will be more like a URL which will be a combination of PeerId and Urn.

rudolfs replied:
It's "Device ID" [now](https://github.com/radicle-dev/radicle-upstream/pull/1206/files#diff-35f363d0d1e254ca6f35454c30107f5fffc88b5da9be72a77e892d1de8e27202R48-R50) as per https://github.com/radicle-dev/radicle-upstream/pull/1206#issuecomment-724317000.

FintanH reviewed (commented):
Nice :+1: All my comments are moot :smile:

abbey-titcomb commented:
> > Add tooltip to Manage Remotes modal?
> 
> Is it these tooltips?
> <img alt="Screenshot 2020-11-09 at 09 37 47" width="50%" src="https://user-images.githubusercontent.com/158411/98518136-45a3a980-226f-11eb-88e0-2700f9742a8d.png">
> 
> If so, I had them there initially, but [had to remove them](https://github.com/radicle-dev/radicle-upstream/pull/1155/files#diff-52c26d24276defc75692d1883dd7f53ffdde278c62b6c6a5ce0fd15c4e904e75L57) after a discussion with @radicle-dev/product.

Ah okay. I added it as a note to be discussed bc it seems we could clear up a lot of confusion around the Manage Remotes section with a tooltip. Will check out the prev discussion on it first.

abbey-titcomb commented:
> Had a preliminary peek. 👀
> 
> Would be great if you would set up commit signing: https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/signing-commits.
> 
> Lmk if you need help with that!

Oops. Thought I did that... Will double check and fix before anymore commits.

abbey-titcomb commented:
> > Add tooltip to Manage Remotes modal?
> 
> Is it these tooltips?
> <img alt="Screenshot 2020-11-09 at 09 37 47" width="50%" src="https://user-images.githubusercontent.com/158411/98518136-45a3a980-226f-11eb-88e0-2700f9742a8d.png">
> 
> If so, I had them there initially, but [had to remove them](https://github.com/radicle-dev/radicle-upstream/pull/1155/files#diff-52c26d24276defc75692d1883dd7f53ffdde278c62b6c6a5ce0fd15c4e904e75L57) after a discussion with @radicle-dev/product.

Realizing that a "tooltip" wasn't what I was trying to actually present. I was thinking of a Helper Box with some copy regarding "adding remotes" that can be closed out, similar to the Git Remote Helper Hint.

abbey-titcomb commented:
After discussing with @juliendonck @cloudhead & @brandonhaslegs, I'm proposing the following changes to app terminology:

- **PeerID** will be renamed to **Device ID**

**Reasoning**: Although PeerID is the name for a keypair tied to a device according to radicle-link, Upstream users will ultimately be using PeerIDs to identify (and manage) their devices when we support for multiple devices. Therefore, it makes more sense to communicate these PeerIDs as they relate to the user's device - hence a _Device ID_. 

- Any **URN** (users and projects) will be renamed to **Radicle ID**. 

**Reasoning:** We believe URNs should be renamed in the app to make them more user-friendly. While a URN *technically* describes the type of identifier that is used to find & identify users/projects in network, it's an acronym that might not make sense to users lacking technical context. On the other hand, other attempts at renaming (i.e. handle, projectID etc...) have abstracted _too much_ away from the core technical concept and introduced connotations that may confuse the user (i.e. a handle on twitter is a name, not a URN).

Radicle ID is user-friendly while maintaining technical accuracy, as the URN is an identifier that can be used to identify users & projects in Radicle - in clearest terms, a Radicle ID. It will be used in the following ways:

_"Share your Radicle ID with others"
"Paste a project's Radicle ID here to find it"
"Add a user's Radicle ID to add them as a remote"_

**Furthermore**, we will remove the user's Radicle ID from onboarding screens & profile pages and replace it with the user's Device ID. Since there is no use for a user's Radicle ID at beta, including it in the app will be confusing to the user as all they will be using is their Device ID. The user's Radicle ID will be introduced along with multi-device support and user discovery. The re-introduction will be paired with some smart documentation and onboarding modals to help users understand the shift from a single Device ID -> many Device IDs + a single Radicle ID. 

While the Device ID can be copied from the user's profile page, it will also be available in the Settings. With the introduction of the user's Radicle ID, it will be replaced on the profile page, but remain in the same place on the Settings page.

I've implemented these copy changes throughout the app, and have updated the TODO's with outstanding items I need support with. Will also update radicle-docs & the glossary to reflect the change. 

Hopefully this will clear everything up for us. Tack!

FintanH commented:
Thanks for the breakdown @abbey-titcomb :clap: Looks good to me, but there's one thing I'd like to flag for consideration.

> Radicle ID is user-friendly while maintaining technical accuracy, as the URN is an identifier that can be used to identify users & projects in Radicle

A URN, later on, will also identify other resources in the network, e.g. issues, code review. Just want to make sure that we think Radicle ID is still appropriate for these types of data -- or any future types of data.

rudolfs commented on ui/Screen/Settings.svelte:
```diff
@@ -144,85 +144,47 @@
             style="color: var(--color-foreground-level-6); margin-bottom: 24px;">
             Enter seed addresses that you’d like to subscribe to here. <a style="color: var(--color-foreground-level-5);" class="typo-link" href="https://radicle.xyz/#upstream-faq"> Learn
-              more about seeds </a>
+              more about seeds. </a>
```
Had this discussion with @brandonhaslegs about whether to put a full stop after a call to action before. I think the decision was to leave it out. The other place where we have a CTA is in the `RemoteHelperHint`.

abbey-titcomb replied:
Hm. If we don't put a `.` then we should probably put a `:` or something no? Nothing looks weird to me. But if you have a strong preference @brandonhaslegs, I don't mind.

rudolfs replied:
Just for context, these are all of the places where we have CTAs as they are in this branch currently:

<img width="50%" alt="Screenshot 2020-11-10 at 14 08 37" src="https://user-images.githubusercontent.com/158411/98678091-541cbe80-235e-11eb-819a-4456344ab2dd.png">
<img width="70%" alt="Screenshot 2020-11-10 at 14 08 39" src="https://user-images.githubusercontent.com/158411/98678094-54b55500-235e-11eb-9530-0e083852ff6c.png">
<img width="50%" alt="Screenshot 2020-11-10 at 14 08 48" src="https://user-images.githubusercontent.com/158411/98678095-54b55500-235e-11eb-88f5-440a1c77eb6f.png">
<img width="50%" alt="Screenshot 2020-11-10 at 14 08 56" src="https://user-images.githubusercontent.com/158411/98678096-554deb80-235e-11eb-9b00-ad7d846d2bb3.png">

I'm only concerned about consistency -- so all the CTAs look the same within the app, I don't really care whether we go with full stop, or not.

> Hm. If we don't put a . then we should probably put a : or something no?

Not sure I follow, could you elaborate?

abbey-titcomb replied:
Cool, I see what you mean. Let's remove the `.` in all CTAs and use that consistently moving forward

rudolfs replied:
https://github.com/radicle-dev/radicle-upstream/pull/1206/commits/e316a945bcd43ccc5137f9878d64bd0d59f4e0d1

rudolfs commented:
Added the peer id to:
- the onboarding success screen
  <img width="606" alt="Screenshot 2020-11-10 at 11 23 42" src="https://user-images.githubusercontent.com/158411/98667356-963e0400-234e-11eb-943e-8a790559627a.png">
- the user profile
  <img width="680" alt="Screenshot 2020-11-10 at 12 05 37" src="https://user-images.githubusercontent.com/158411/98667417-a7871080-234e-11eb-8ec7-cb8aea167146.png">

abbey-titcomb commented:
> Thanks for the breakdown @abbey-titcomb 👏 Looks good to me, but there's one thing I'd like to flag for consideration.
> 
> > Radicle ID is user-friendly while maintaining technical accuracy, as the URN is an identifier that can be used to identify users & projects in Radicle
> 
> A URN, later on, will also identify other resources in the network, e.g. issues, code review. Just want to make sure that we think Radicle ID is still appropriate for these types of data -- or any future types of data.

Yeah, great point. IMO, it holds up:

"Send me #1209's Radicle ID" or even just shortened to "Send me #1209's ID". I can even see Radicle ID not being used in app copy so much since people just will directly link it in their discussions/conversations.

juliendonck reviewed (dismissed):
Looks great. Nice work @abbey-titcomb & co :)

FintanH reviewed (approved):
Ja, looks good to me
